sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts

```diff
@@ -0,0 +1,103 @@
+import { Controller, Get, Param } from "@nestjs/common";
+import BaseController from "../BaseController";
+import { AllowAuthorizedParty, Groups } from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { ClientTypeBaseRoute } from "../../types";
+import { ApiOkResponse, ApiTags } from "@nestjs/swagger";
+import { UserGroups } from "../../auth/user-groups.enum";
+import {
+  StudentAppealService,
+  StudentAssessmentService,
+  StudentScholasticStandingsService,
+} from "../../services";
+import {
+  AssessmentTriggerType,
+  ScholasticStandingStatus,
+  StudentAppealStatus,
+} from "../../database/entities";
+import { RequestAssessmentSummaryDTO } from "./models/assessment.dto";
+import { AssessmentHistory } from "../../services/student-assessment/student-assessment.models";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("assessment")
+@ApiTags(`${ClientTypeBaseRoute.AEST}-assessment`)
+export class AssessmentAESTController extends BaseController {
+  constructor(
+    private readonly studentAppealService: StudentAppealService,
+    private readonly studentScholasticStandingsService: StudentScholasticStandingsService,
+    private readonly studentAssessmentService: StudentAssessmentService,
+  ) {
+    super();
+  }
+
+  /**
+   * Method to get all requested assessments for a student
+   * application, i.e, this will fetch the combination of
+   * pending and denied student appeal and scholastic
+   * standings for an application.
+   * @param applicationId, application number.
+   * @returns RequestAssessmentSummaryDTO list.
+   */
+  @Get("application/:applicationId/requests")
+  @ApiOkResponse({
+    description: "Requested assessment for the application found.",
+  })
+  async getRequestedAssessmentSummary(
+    @Param("applicationId") applicationId: number,
+  ): Promise<RequestAssessmentSummaryDTO[]> {
+    const [studentAppealQuery, studentScholasticStandingsQuery] =
+      await Promise.all([
+        this.studentAppealService.getPendingAndDeniedAppeals(applicationId),
+        this.studentScholasticStandingsService.getPendingAndDeniedScholasticStanding(
+          applicationId,
+        ),
+      ]);
+
+    let requestedAssessments = [
+      ...studentAppealQuery.map((appeals) => ({
+        submittedDate: appeals.submittedDate,
+        status: appeals.status,
+        triggerType: AssessmentTriggerType.StudentAppeal,
+      })),
+      ...studentScholasticStandingsQuery.map((scholasticStanding) => ({
+        submittedDate: scholasticStanding.submittedDate,
+        status: scholasticStanding.status,
+        triggerType: AssessmentTriggerType.ScholasticStandingChange,
+      })),
+    ];
+
+    // submitted date sorting
+    requestedAssessments = requestedAssessments.sort((first, second) =>
```

Comment:
I am not able to follow here. Why are we sorting in dataset instead of query?

Reply:
I guess because the data-set is an union of 2 tables. Am I right?

Reply:
If a db view approach can make it simple, may be we can put that as techical debt.

Reply:
please see the service functions, Request datatable need the combination of student appeals table and scholastic standing table

Reply:
> If a db view approach can make it simple, may be we can put that as techical debt.

Yes, I also would say that it is worth taking a look at the DB view approach. We have this opportunity in the upcoming ticket for pending assessments requests where similar results must be paginated.